types/time.go

```diff
@@ -1712,6 +1712,17 @@ func ParseDate(sc *stmtctx.StatementContext, str string) (Time, error) {
 	return ParseTime(sc, str, mysql.TypeDate, MinFsp)
 }
 
+// ParseTimeFromYear parse a `YYYY` formed year to corresponded Datetime type.
+// year must be in the range [MinYear, MaxYear].
```

Comment:
I do not understand why removing the [MinYear, MaxYear] check?

Reply:
`create table t (a datetime, b year);
insert into t values("2020-11-11 11:11:11", 2019);  success
insert into t values("2020-11-11 11:11:11", 2222); fail. 
`
 At least one of the column is type Year, the data is checked when inserting, it should be guaranteed that invalid year which out of range cannot be inserted successfully.

So we do not need to check it is out of range. IMO, it's impossible ...

Reply:
I mean, the inserted year, must already in the range [MinYear, MaxYear].